kate/src/lib.rs

```diff
@@ -100,55 +210,33 @@ pub fn padded_len(len: u32, chunk_size: u32) -> u32 {
 	iec_9797_1_len + pad_to_chunk_extra
 }
 
-#[derive(Clone, Copy, PartialEq, Eq, Debug)]
+#[cfg_attr(feature = "std", derive(Debug))]
+#[derive(Clone, Copy, PartialEq, Eq, Constructor)]
 pub struct BlockDimensions {
 	pub rows: BlockLengthRows,
 	pub cols: BlockLengthColumns,
-	pub chunk_size: u32,
+	pub chunk_size: NonZeroU32,
 }
 
 impl BlockDimensions {
-	pub fn size(&self) -> usize {
-		self.rows
-			.0
-			.saturating_mul(self.cols.0)
-			.saturating_mul(self.chunk_size) as usize
-	}
-
-	pub fn new<R, C>(rows: R, cols: C, chunk_size: u32) -> Self
-	where
-		R: Into<BlockLengthRows>,
-		C: Into<BlockLengthColumns>,
-	{
-		Self {
-			rows: rows.into(),
-			cols: cols.into(),
-			chunk_size,
-		}
+	pub fn size(&self) -> Option<usize> {
+		let rows_cols = self.rows.0.checked_mul(self.cols.0)?;
+		let rows_cols_chunk = rows_cols.checked_mul(self.chunk_size.get())?;
+		usize::try_from(rows_cols_chunk).ok()
```

Comment:
Another option would be to move all the checks into a constructor and make the fields private. This would make the interface easier to reason with because it would be impossible to construct a `BlockDimensions` object with too many rows and cols.